src/bls/bls_worker.cpp

```diff
@@ -250,17 +248,18 @@ struct Aggregator {
             delete rem[i];
         }
         doneCallback(r);
-
-        delete this;
     }
 
-    void AsyncAggregateAndPushAggQueue(std::shared_ptr<std::vector<const T*> >& vec, size_t start, size_t count, bool del)
+    void AsyncAggregateAndPushAggQueue(std::shared_ptr<std::vector<const T*>> vec, size_t start, size_t count, bool del)
```

Comment:
Why is this vec (and below) no longer passed by reference?

I guess it is a reference to a shared_ptr, so likely makes sense to just pass by value. Is that the intention?

Reply:
You are right, only the lambda (line 257) has to capture `vec` by value as the lambda is going to be passed to another thread and we want to share the ownership of `vec`. I will change them back to shared_ptr& to avoid useless increments of the refcounter. Also, I will mark the shared_ptr& param as `const`.